### PR TITLE
Remove redundant join-reordering from JoinPlanBuilder

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

This removes the join-ordering as part of the JoinPlanBuilder because the logic is redundant with `optimizer_eliminate_cross_join`. 

This code was already removed  when https://github.com/crate/crate/commit/1cb03b79e0c5b4d49b24ec8fefaa45642bf3db33 was implemented. The sql-logic tests revealed a bug related to hash-joins hidden by the ordering and the code was readded https://github.com/crate/crate/commit/288c4b7a9904708df9a7581787c895de413518d2. The related bug was then fixed in https://github.com/crate/crate/commit/11b16abf53d9ce083d05a415598846a10667ccbd. 

Sql-logic tests are passing now on this pr, so it's safe to finally remove the code.

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java [hunk 1]
```diff
@@ -171,7 +174,8 @@ static LogicalPlan reorder(JoinGraph graph, List<LogicalPlan> order) {
                         errorMessage.add(relationName.fqn());
                     }
                 }
-                throw new InvalidArgumentException("JoinPlan cannot be built with the provided order " + errorMessage);
+                LOGGER.trace("JoinPlan cannot be built with the provided order {}", errorMessage);
+                return null;
```
Comment (mkleen) [inline server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java:178]: If we fail on reordering, let's skip this rule by returning null.

Comment (matriv) [reply server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java:178]: yep, I agree, but I think we should use: `LoggingOptimizerTracer` and not a specific logger for this class, wdyt?

**Review by matriv — APPROVED** [review]
Thx for keeping track and cleaning up code!
I've ran the sqllogic tests one more time on this branch just to be on the safe side.

Left a comment regarding the logger.

Thx again! LGTM

File: server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java (hunk 1 above)
Comment (mkleen) [reply server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java:178]: `OptimizerTracer` is used to track the progress of the optimizer not for error messages.

Comment (mkleen) [reply server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java:178]: So `LoggingOptimizerTracer` tells us which rule was executed and what progress/transformation was made on the query plan.

Comment (matriv) [reply server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java:178]: ok, I see, makes sense to leave as is, thx!

**Pull request merged**